src/util/gas-factory-helpers.ts

```diff
@@ -362,11 +370,12 @@ export function initSwapRouteFromExisting(
   v2PoolProvider: IV2PoolProvider,
   v3PoolProvider: IV3PoolProvider,
   portionProvider: IPortionProvider,
-  quoteGasAdjusted: CurrencyAmount<Currency>,
+  quoteGasAdjusted: CurrencyAmount,
```

Comment:
nit: keeping generic does have benefit I think, and `Currency` is a super type of `NativeCurrency` and `Token` right?

Reply:
Seems like our local `CurrencyAmount` is just this:
```
export class CurrencyAmount extends CurrencyAmountRaw<Currency> {}
```
And I initially tried keeping the generic CurrencyAmount here but was getting weird type issues when dealing with wrapped native currency